sdk/eventhub/Azure.Messaging.EventHubs/src/ExponentialRetry.cs

```diff
@@ -76,7 +117,19 @@ public sealed class ExponentialRetry : Retry
         /// <returns>A hash code for this instance, suitable for use in hashing algorithms and data structures like a hash table.</returns>
         ///
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public override int GetHashCode() => base.GetHashCode();
+        public override int GetHashCode()
+        {
+            int hash = 17;
+
+            unchecked
+            {
+                hash = (hash * 23) + _minimumBackoff.GetHashCode();
```

Comment:
Use HashCodeBuilder from Azure.Core.

Reply:
Can't.  It's an internal type.

Reply:
We share it as a source reference.

Reply:
I'm against incorporating raw source from other projects.   If it's meant to be shared, I'd advocate for it being a public type or abstracted into a different library of shared primitives.

Reply:
As I said, I'm happy to do that but there are reasons that prevent us from doing that. One being we don't want it as public API.

Reply:
That would require a dependency from our library to the specific version of core that has that API. 

The same is true if we have a dependency on a given assembly -- if that assembly contains a lot of low-level helpers, there is a good chance that multiple libraries would depend on different versions of that assembly, leading to problems.

Reply:
@AlexGhiondea:  I see the point that you're making, though I'm really afraid of the unintended consequences of "oh, this file was in my project, I own it and it is safe to change.  I concede the versioning issue with respect to dependencies, however.  

In the case that we wish to have a set of source that is truly shared among projects in the repository,  can we then have these live in a specific namespace that indicates "This is shared source and making changes will have a cascade effect?"    As it stands today, having these live in `Core` implies ownership by `CoreI` rather than something intended for sharing.  It feels like we're mixing concerns.

Reply:
We can move the files to a shared folder (if they are not there already) and we should be linking the file in, instead of copying in.

Reply:
> We can move the files to a shared folder (if they are not there already) and we should be linking the file in, instead of copying in.

That's how it's being done right now.

Reply:
The internalization of retry equality makes the use of `HashCodeBuilder` here a moot point, but I think the discussion of sharing source is worth continuing regardless.

Reply:
> > We can move the files to a shared folder (if they are not there already) and we should be linking the file in, instead of copying in.
> 
> That's how it's being done right now.

Not quite at the level that I'm asking for.  My intent is that shared folder lives outside the scope of any individual project;  currently that lives under `Core` which implies ownership of those files by the `Core` project and that they're "shared within the project", if one assumes the popular use.   For example, assembly information shared among projects in a solution is often kept in a solution-level `Shared` folder.

I'm proposing moving those into a namespace outside of Core  (`Azure.Shared`, maybe?)  and to a location in the repository outside of any individual project (`/src/ClientLibraryCommon`, or similar?)

Reply:
I don't have a strong opinion of where to put shared source as long as it's being used.

Reply:
I would be fine with a location outside of the Core library.

Reply:
That would mitigate my concerns;  with the source intended to be shared in a location and namespace that clearly signals that intent, it becomes the responsibility of each project that references it to clearly denote the ownership.   

Are there any follow-ups needed from this discussion?  Issues to open or other actions?